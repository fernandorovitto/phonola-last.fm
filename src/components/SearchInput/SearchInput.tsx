import { IconButton } from '@material-ui/core';
import Input from '@material-ui/core/Input';
import React from 'react';
import SearchIcon from '@material-ui/icons/Search';
import styled from 'styled-components';

interface PropsFromDispatch {
  search: (value: string) => void;
}

export type IProps = PropsFromDispatch;

interface IState {
  value: string;
}

class SearchInput extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      value: '',
    };
  }

  private _handleChange = (
    event: React.ChangeEvent<HTMLInputElement>,
  ): void => {
    const cleanValue = event.target.value.replace(/[^a-zA-Z0-9 ]/g, '');
    this.setState({ value: cleanValue });
  };

  private _keyPressed = (event: React.KeyboardEvent): void => {
    if (event.key === 'Enter') {
      this._search();
    }
  };

  private _search = (): void => {
    this.props.search(this.state.value);
    this.setState({ value: '' });
  };

  public render(): React.ReactNode {
    return (
      <SearchBar>
        <StyledInput
          className="input"
          id="searchInput"
          inputProps={{ 'aria-label': 'description' }}
          onChange={this._handleChange}
          onKeyPress={this._keyPressed}
          placeholder="Search for your artist..."
          type="text"
          value={this.state.value}
        />
        <IconButtonContainer>
          <IconButton aria-label="search" size="small" onClick={this._search}>
            <SearchIcon />
          </IconButton>
        </IconButtonContainer>
      </SearchBar>
    );
  }
}

export default SearchInput;

// Styled components

const SearchBar = styled('div')`
  background-color: white;
  display: flex;
`;

const IconButtonContainer = styled('div')`
  height: 30px;
  right: 0;
  top: 0;
`;

const StyledInput = styled(Input)`
  font-size: 14px;
  margin-left: 5px;
  padding: 0;
`;
