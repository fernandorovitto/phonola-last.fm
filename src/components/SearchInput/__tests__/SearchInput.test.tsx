import 'jest-styled-components';
import SearchInput, { IProps } from '../SearchInput';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Test for SearchInput Component', () => {
  test('Should render the component', () => {
    const props: IProps = {
      search: jest.fn(),
    };
    const tree = renderer.create(<SearchInput {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
