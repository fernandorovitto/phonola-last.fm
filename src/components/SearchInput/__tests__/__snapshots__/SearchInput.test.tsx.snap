// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for SearchInput Component Should render the component 1`] = `
.c0 {
  background-color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  height: 30px;
  right: 0;
  top: 0;
}

.c1 {
  font-size: 14px;
  margin-left: 5px;
  padding: 0;
}

<div
  className="c0"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline c1 input"
    onClick={[Function]}
    onKeyPress={[Function]}
  >
    <input
      aria-label="description"
      className="MuiInputBase-input MuiInput-input"
      id="searchInput"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search for your artist..."
      type="text"
      value=""
    />
  </div>
  <div
    className="c2"
  >
    <button
      aria-label="search"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
